libtiledbsoma/src/utils/util.cc

```diff
@@ -81,20 +81,42 @@ std::vector<uint8_t> cast_bit_to_uint8(ArrowSchema* schema, ArrowArray* array) {
             schema->format));
     }
 
-    const void* data;
+    uint8_t* data;
     if (array->n_buffers == 3) {
-        data = array->buffers[2];
+        data = (uint8_t*)array->buffers[2];
```

Comment:
this code doesn't seem to consider the validity buffer when converting the bit vector to the byte vector.  I'm not entirely clear on what are semantics should be, but based on Pythonic conventions, an invalid (nulled) position would commonly be cast to `false`.

I suspect (but wanted to confirm) it is all handled upstream (e.g., the validity buffer is written separately, and takes precedence over any value in `casted` data.

Reply:
This is done upstream: https://github.com/single-cell-data/TileDB-SOMA/blob/viviannguyen/sc-61239/c-tiledbsoma-soma-write-will-corrupt-data/libtiledbsoma/src/soma/managed_query.cc#L1012-L1030